presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxHistoricalData.java

```diff
@@ -27,24 +30,36 @@
 import java.util.Set;
 
 import static com.google.common.base.Preconditions.checkArgument;
-import static java.util.stream.Collectors.toSet;
+import static io.prestosql.plugin.jmx.JmxUtils.toPattern;
+import static java.util.Locale.ENGLISH;
+import static javax.management.ObjectName.WILDCARD;
 
 public class JmxHistoricalData
 {
     private final Set<String> tables;
     private final Map<String, EvictingQueue<List<Object>>> tableData = new HashMap<>();
 
     @Inject
-    public JmxHistoricalData(JmxConnectorConfig jmxConfig)
+    public JmxHistoricalData(JmxConnectorConfig jmxConfig, MBeanServer mbeanServer)
     {
-        this(jmxConfig.getMaxEntries(), jmxConfig.getDumpTables());
+        this(jmxConfig.getMaxEntries(), jmxConfig.getDumpTables(), mbeanServer);
     }
 
-    public JmxHistoricalData(int maxEntries, Set<String> tableNames)
+    public JmxHistoricalData(int maxEntries, Set<String> tableNames, MBeanServer mbeanServer)
     {
-        tables = tableNames.stream()
-                .map(tableName -> tableName.toLowerCase(Locale.ENGLISH))
-                .collect(toSet());
+        ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        try {
```

Comment:
shouldn't this `try` be inside the `for`? Do you have a test where this exception is thrown? It would be nice to have a test with several tables where one of them (in the middle) is incorrect?

or we could simply fail the Presto by not ignoring the exception